esiclient/tests/unit/v1/test_node_network.py

```diff
@@ -469,6 +473,33 @@ def test_take_action_no_port_or_network_or_trunk_exception(self):
             'ERROR: You must specify either network, port, or trunk',
             self.cmd.take_action, parsed_args)
 
+    # @mock.patch('esiclient.utils.get_full_network_info_from_port',
+    #             return_value=(["test_network_invalid"], ["node1-port"],
+    #                           ["2.2.2.2"]),
+    #             autospec=True)
+    def test_take_action_invalid_network_exception(self):
+        self.app.client_manager.baremetal.node.get.\
+            return_value = self.node
+        self.app.client_manager.baremetal.port.list.\
+            return_value = [self.port1]
+        self.app.client_manager.network.find_network.\
+            return_value = self.network
```

Comment:
So what's happening here is that even though you are passing in 'test_network_invalid' - a network that doesn't exist - you are also telling `find_network` to always return a network. Because of that, this test will always think that the network exists. What you want to do is set this return_value to None instead.